python/tskit/provenance.py

```diff
@@ -132,3 +135,44 @@ def validate_provenance(provenance):
         jsonschema.validate(provenance, schema)
     except jsonschema.exceptions.ValidationError as ve:
         raise exceptions.ProvenanceValidationError from ve
+
+
+def provenance_command_JSON(exclude_args=[]):
+    """
+    This function can be called within a function and will encode the arguments used
+    to call that function into a JSON string, which is returned.
+
+    The exclude_args argument allows arguments to be excluded from the prevenance info
+    such as deprecated aliases for other arguments (the `self` argument of methods is
+    always excluded).
+
+    It can be used to e.g. serialize the arguments to functions like `simplify`
+    so they can be stored in a provenance field. It requires the arguments to be JSON
+    serializable (and uses np.array_repr to serialize numpy arrays, which are not
+    normally serializable by json).
+    """
+    def serialize_numpy_array(arg):
```

Comment:
Pull this out into it's own function I'd say, so it can be tested independently.

Reply:
Done - no tests